File: Backend/QuanLyKhamBenhAPI/Models/Doctor.cs
using System;
using System.Collections.Generic;

namespace QuanLyKhamBenhAPI.Models;

public partial class Doctor
{
    public int DoctorId { get; set; }

    public string Name { get; set; } = null!;

    public string? Phone { get; set; }

    public int? SpecialtyId { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();

    public virtual Specialty? Specialty { get; set; }

    public virtual ICollection<UserAccount> UserAccounts { get; set; } = new List<UserAccount>();

    public virtual ICollection<WorkShift> WorkShifts { get; set; } = new List<WorkShift>();
}

File: FrontendMobile/lib/models/doctor.dart
class Doctor {
  final int doctorId;
  final String name;
  final String phone;
  final Specialty? specialty;

  Doctor({
    required this.doctorId,
    required this.name,
    required this.phone,
    this.specialty,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      doctorId: json['doctorId'] ?? 0,
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      specialty: json['specialty'] != null 
          ? Specialty.fromJson(json['specialty']) 
          : null,
    );
  }
}

class Specialty {
  final int specialtyId;
  final String name;

  Specialty({
    required this.specialtyId,
    required this.name,
  });

  factory Specialty.fromJson(Map<String, dynamic> json) {
    return Specialty(
      specialtyId: json['specialtyId'] ?? 0,
      name: json['name'] ?? '',
    );
  }
}
